### PR TITLE
Support for C11/C++11 syntax

### PR DESCRIPTION
**Pull request by Tosainu**
This PR adds support for C11/C++11 syntax.

* Some keywords
```cpp
// C11
_Alignas _Alignof _Atomic _Generic _Imaginary _Noreturn _Static_assert _Thread_local
char16_t char32_t

// C++11
alignas alignof constexpr decltype noexcept static_assert thread_local

// missing C++ keywords
bool try
```

* UTF-8, UTF-16, UTF-32 encoding prefixes (C11/C++11)
```c
char s1[] = u8"UTF-8 string";
char16_t s2[] = u"UTF-16 string";
char32_t s3[] = U"UTF-32 string";
```

* Raw string literal (C++11)
```cpp
char s1[] = R"(\w+ \d+)";
char s2[] = R"delim(\w+ \d+)delim";
char16_t s3[] = uR"(\w+ \d+)";
```

### CODE REVIEW

**Comment by jneen** [comment]
Hi, thanks!  Could you add some examples to `spec/visual/samples/cpp` and check them out with `bundle exec rackup`?

**Comment by Tosainu** [comment]
Thanks for the feedback!
I fixed the code a little bit and added examples. How does it look?

File: lib/rouge/lexers/cpp.rb [hunk 1]
```diff
@@ -24,6 +24,15 @@ def self.keywords
...
           throws typeid typename using virtual
+
+          alignas alignof constexpr decltype noexcept static_assert
+          thread_local try
+        ))
+      end
+
+      def self.keywords_type
+        @keywords_type ||= super + Set.new(%w(
+          bool
```
Comment (zrax) [inline lib/rouge/lexers/cpp.rb]: Please also add `nullptr_t`, and then I can close #355 :)

**Comment by jneen** [comment]
Hey @Tosainu, thanks for the work, this looks good :]

**Comment by Tosainu** [comment]
Thanks for the merge! :star2:

File: lib/rouge/lexers/cpp.rb (hunk 1 above)
Comment (Tosainu) [inline lib/rouge/lexers/cpp.rb]: Sorry to my late reply. but I think `nullptr_t` is not C++ keyword...

**Comment by jneen** [comment]
Released in v1.11.0.

**Pull request merged**